### PR TITLE
json directory

### PR DESCRIPTION
**Pull request by tranma**
for testing and stuff

### CODE REVIEW

**Comment by thsutton** [comment]
I think the json-directory executable should probably take another (mandatory?) argument: the directory path for the JSON files to live in.

File: datasource/json-directory.hs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+--   Prints the filename to `stdout`.
+--
+jsonCreate :: FilePath -> IO ExitCode
+jsonCreate base = do
+  k      <-  T.pack . take 64 . randomRs ('a', 'z')
+         <$> newStdGen
+  exists <- fileExist (mkJSONFilename base k)
+  if   exists
+  then jsonCreate base
+  else do print k
```
Comment (thsutton) [inline datasource/json-directory.hs]: This should read stdin and write it into the new file, right?

File: datasource/json-directory.hs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+-- | Update a 'Document' with input from stdin.
+--   Assuming the input is already in JSON and makes no attempt to check. It
+--   is synchronised's repsonsiblity.
+--
+jsonUpdate :: FilePath -> FilePath -> IO ExitCode
+jsonUpdate base filename = do
+  doc <- B.getContents
+  handle (\(_ :: SomeException) -> return (ExitFailure 1))
+         (do B.writeFile (base </> filename )doc
```
Comment (thsutton) [inline datasource/json-directory.hs]: space

File: datasource/json-directory.hs (hunk 1 above)
Comment (tranma) [inline datasource/json-directory.hs]: no? by creating you only get a "primary key" which will get written when you update.

Comment (thsutton) [inline datasource/json-directory.hs]: https://github.com/anchor/synchronise/blob/master/lib/Synchronise/DataSource.hs#L125

**Pull request merged**